test/jdk/javax/sound/midi/Sequencer/MetaCallback.java

```diff
@@ -101,13 +106,17 @@ static ShortMessage MidiMsg3(int a, int b, int c) {
         }
     }
     void start() {sequencer.start();}
-    void stop() {sequencer.stop();}
+    void stop() {
+        sequencer.stop();
+        sequencer.close();
+    }
 
     public void meta(MetaMessage msg) {
         System.out.println(""+metaCount+": got "+msg);
         if (msg.getType() == 0x2F) {
             finished = true;
-        } else if (msg.getData().length > 0 && msg.getType() == 1) {
+        } else if (msg.getData().length > 0 && msg.getType() == 1 && !received.contains(msg)) {
+            received.add(msg);
```

Comment:
Do we not need to do seq.stop() before close() in l136

Reply:
If this code is needed can we check first where we create duplicate events?

Reply:
> Do we not need to do seq.stop() before close() in l136

On that line we never call start() on the sequencer so calling stop() is redundant.

Reply:
ok, understood...

Reply:
Still have a question why we skip this duplicates and do not consider them as a bug?

Reply:
> Still have a question why we skip this duplicates and do not consider them as a bug?

I do not have enough information to act on this issue because while not being problem listed it only occurred twice on CI system and in more than a thousand runs i was able to reproduce it once on my local system and unable to reproduce it again with all the output i've added to the code. So what i am planning to do is to leave workaround code in place but for now i will disable it and instead use it as an indicator of the problem and add a debugging output so if this issue will reappear we will have better understanding of the potential problem.

Reply:
Thank you. That looks great.